circuit-lib/circuit-lib.js/src/elgamal-babyjubjub/elgamal.ts

```diff
@@ -0,0 +1,287 @@
+import { AffinePoint } from "@noble/curves/abstract/curve";
+import { ExtPointType } from "@noble/curves/abstract/edwards";
+import { babyJub as CURVE } from "./babyjubjubConfig";
+import { encode } from "./decode";
+import assert from "assert";
+import * as crypto from "crypto";
+
+const createBlakeHash = require("blake-hash");
+const ff = require("ffjavascript");
+const Scalar = ff.Scalar;
+
+type SnarkBigInt = bigint;
+type PrivKey = bigint;
+type PubKey = ExtPointType;
+type BabyJubAffinePoint = AffinePoint<bigint>;
+type BabyJubExtPoint = ExtPointType;
+
+/**
+ * A private key and a public key
+ */
+interface Keypair {
+  privKey: PrivKey;
+  pubKey: PubKey;
+}
+
+// The BN254 group order p
+const SNARK_FIELD_SIZE: SnarkBigInt = BigInt(
+  "21888242871839275222246405745257275088548364400416034343698204186575808495617",
+);
+// Textbook Elgamal Encryption Scheme over Baby Jubjub curve without message encoding
+const babyJub = CURVE.ExtendedPoint;
+
+/**
+ * Returns a BabyJub-compatible random value. We create it by first generating
+ * a random value (initially 256 bits large) modulo the snark field size as
+ * described in EIP197. This results in a key size of roughly 253 bits and no
+ * more than 254 bits. To prevent modulo bias, we then use this efficient
+ * algorithm:
+ * http://cvsweb.openbsd.org/cgi-bin/cvsweb/~checkout~/src/lib/libc/crypt/arc4random_uniform.c
+ * @return A BabyJub-compatible random value.
+ * @see {@link https://github.com/privacy-scaling-explorations/maci/blob/master/crypto/ts/index.ts}
+ */
+function genRandomBabyJubValue(): bigint {
+  // Prevent modulo bias
+  //const lim = BigInt('0x10000000000000000000000000000000000000000000000000000000000000000')
+  //const min = (lim - SNARK_FIELD_SIZE) % SNARK_FIELD_SIZE
```

Comment:
can we remove these comments?

Reply:
Removed after replacing code with imported functions from [maci-crypto](https://www.npmjs.com/package/maci-crypto)